src/plots/cartesian/position_defaults.js

```diff
@@ -10,8 +10,10 @@ module.exports = function handlePositionDefaults(containerIn, containerOut, coer
     var overlayableAxes = options.overlayableAxes || [];
     var letter = options.letter;
     var grid = options.grid;
+    var shift = containerIn.shift;
```

Comment:
We should try not to use `containerIn.shift` directly which could cause issues e.g. when `shift` being set via `layout.template.yaxis.shift`. 
Instead we should let `coerce` functions handle that for us.

Reply:
I'm not fully understanding how to apply `coerce` here. What would be the best way to access the value of `shift` as it is defined (or `undefined`) in the figure definition?